From 31b6231f420d6e458416622ac6a04f0c3c4fd770 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:24:37 -0400
Subject: [cpufreq] propagate error returned by kobject_move()

Message-id: <1409235879-1283-184-git-send-email-prarit@redhat.com>
Patchwork-id: 89489
O-Subject: [RHEL7.1 PATCH BZ 1134639 183/185] cpufreq: propagate error returned by kobject_move()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit 41dfd908fc09e0e57f8f44fdcf543348cd5f249c
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Thu Jul 17 10:48:27 2014 +0530

    cpufreq: propagate error returned by kobject_move()

    We are returning -EINVAL instead of the error returned from kobject_move() when
    it fails. Propagate the actual error number.

    Also add a meaningful print when sysfs_create_link() fails.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index d02e730..7a432c1 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -1320,10 +1320,12 @@ static int cpufreq_nominate_new_policy_cpu(struct cpufreq_policy *policy,
  ret = kobject_move(&policy->kobj, &cpu_dev->kobj);
  if (ret) {
   pr_err("%s: Failed to move kobj: %d\n", __func__, ret);
-  ret = sysfs_create_link(&cpu_dev->kobj, &policy->kobj,
-     "cpufreq");
 
-  return -EINVAL;
+  if (sysfs_create_link(&cpu_dev->kobj, &policy->kobj, "cpufreq"))
+   pr_err("%s: Failed to restore kobj link to cpu:%d\n",
+          __func__, cpu_dev->id);
+
+  return ret;
  }
 
  return cpu_dev->id;
-- 
1.7.1